hack/test-go.sh

```diff
@@ -303,6 +309,9 @@ reportCoverageToCoveralls() {
   fi
 }
 
+# Start etcd
+kube::etcd::start
```

Comment:
> (in particular - unit tests can "communicate" with each other via this etcd instance).
We could always just randomize the key used to test as they all need a /prefixkey...?  

> And even better would be to start it not in a separate process, but just link the etcd code to our tests and start it e.g. as a separate goroutine. 
If we did this we'd have to be a stickler on the cleanup mechanics.